server/datastore/mysql/queries.go

```diff
@@ -606,3 +636,66 @@ func (ds *Datastore) CleanupGlobalDiscardQueryResults(ctx context.Context) error
 
 	return nil
 }
+
+// IsSavedQuery returns true if the given query is a saved query.
+func (ds *Datastore) IsSavedQuery(ctx context.Context, queryID uint) (bool, error) {
+	stmt := `
+		SELECT saved
+		FROM queries
+		WHERE id = ?
+	`
+	args := []interface{}{queryID}
+	results := []bool{}
+	if err := sqlx.SelectContext(ctx, ds.reader(ctx), &results, stmt, args...); err != nil {
```

Comment:
I believe this may return ErrNoRows.  You may want to use a single bool var and check for that err instead of using the len check

Reply:
Yes, changed to `GetContext` to get the `ErrNoRows` functionality.